pycloudlib/azure/security_types.py

```diff
@@ -0,0 +1,136 @@
+"""Azure Security Types Classes."""
+import abc
+import re
+from enum import Enum
+from typing import Any, Dict
+
+from pycloudlib import util
+
+
+class AzureCVMOSDiskEncryption(Enum):
+    """Represents Azure OS disk encryption types."""
+
+    VM_GUEST_STATE_ONLY = "VMGuestStateOnly"
+    DISK_WITH_VM_GUEST_STATE = "DiskWithVMGuestState"
+
+
+class SecurityType(abc.ABC):
+    """Base Azure Security Type Class."""
+
+    @abc.abstractmethod
+    def configure_vm_params(
+        self, vm_params: Dict[str, Any], **kwargs
```

Comment:
As of now, the only kwarg that is used is `os_disk_encryption` in `AzureConfidentialVMSecurityType`. What about being explicit in the interface and keep it simple, doing something in the following lines:

```suggestion
        self, vm_params: Dict[str, Any], os_disk_encryption: Optional[AzureCVMOSDiskEncryption] = None
```

Reply:
I agree but I think we will  need this `kwargs` later for any change in the security types configuration, considering MS api changes a lot.